codegen/test/common.hpp

```diff
@@ -1,25 +1,46 @@
 #pragma once
+
+#include "ck/host/headers.hpp"
+#include "ck/host/stringutils.hpp"
+#include <rtc/compile_kernel.hpp>
+#include <rtc/hip.hpp>
+#include <test.hpp>
 #include <algorithm>
 #include <cmath>
+#include <fstream>
 #include <iterator>
 #include <numeric>
 #include <random>
-#include <test.hpp>
-#include <rtc/compile_kernel.hpp>
-#include <rtc/hip.hpp>
-#include <fstream>
+#include <unordered_set>
+
+// NOLINTNEXTLINE
+const char* const ck_content_wrapper = R"__ck__(
+${content}
+)__ck__";
+
+template <class P>
+inline std::string content_wrapper(P p)
+{
+    return ck::host::InterpolateString(ck_content_wrapper,
+                                       {{"content", std::string{p.data(), p.size()}}});
+}
 
-std::vector<rtc::src_file> get_headers_for_test()
+inline std::vector<rtc::src_file> create_headers_for_test()
 {
+    auto ck_headers = ck::host::GetHeaders();
     std::vector<rtc::src_file> result;
-    auto hs = ck::host::GetHeaders();
-    std::transform(
-        hs.begin(), hs.end(), std::back_inserter(result), [&](const auto& p) -> rtc::src_file {
-            return {p.first, p.second};
-        });
+    std::transform(ck_headers.begin(), ck_headers.end(), std::back_inserter(result), [](auto& p) {
+        return rtc::src_file{p.first, content_wrapper(p.second)};
```

Comment:
Remove `content_wrapper`, just do `rtc::src_file{p.first, p.second}`.

Reply:
When I remove the wrapper, `hiprtcCreateProgram` fails with `HIPRTC_ERROR_INVALID_INPUT`. I have to take a closer look to see what's going on.

Reply:
For some reason, hipRTC fails if there's no whitespace before the contents. I implemented a workaround for now.